cvat-canvas/src/typescript/canvasModel.ts

```diff
@@ -250,15 +258,41 @@ export class CanvasModelImpl extends MasterImpl implements CanvasModel {
             this.data.canvasSize.width / FrameZoom.MIN,
         ));
 
-        this.notify(UpdateReasons.FIT_CANVAS);
-        this.notify(UpdateReasons.OBJECTS);
+        this.notify(UpdateReasons.FITTED_CANVAS);
+        this.notify(UpdateReasons.OBJECTS_UPDATED);
+    }
+
+    public dragCanvas(enable: boolean): void {
+        if (enable && this.data.mode !== Mode.IDLE) {
```

Comment:
What about moving this condition in separate function? Like so:
```javascript
public checkIfCanvasIsBusy(enable: boolean): void {
        if (enable && this.data.mode !== Mode.IDLE) {
            throw Error(`Canvas is busy. Action: ${this.data.mode}`);
        }
}
```

Reply:
I am not sure it makes sense. 
Yes, this condition is repeated twice, but other methods of ``cvat-canvas`` also check if canvas is busy, but using other conditions.